### ISSUE TITLE
Getting JSON.parse error in teamsFx internal file when the conversation is initiated from the MS Teams

### ISSUE THREAD

**Issue by bhaskar-012024** [OfficeDev/TeamsFx#12887]
Key Components Used:
1. Azure Bot: For creation of the Bot which is referred in the Teams app. Here the target was the Azure WebApp. Any message to the bot will go through the Azure Bot and then to the Azure WebApp
2.  Azure WebApp: This is the place where my Node.js server resides and here we are using teams toolkit(teamsFx) for developed logic related to the bot interaction. It is Linux based server.

Recently we started getting errors when the user interacts with the Bot via Teams app. Interesting thing is we haven't made any changes to the Azure Bot or WebApp or any component in the structure recently but suddenly we stated seeing below error that is refereeing to a JSON.parse statement in the @microsoft/teamsfx/dist/index.node.cjs.js internal file, which obviously we haven't directly called in any of our code. Previously it is working perfectly fine, and we don't have any issue but not sure why we are receiving this error now without any change in configuration or code.

When we test the same bot directly from the Azure portal inside Azure Bot> Settings> Test in Web Chat it is working perfectly fine but we are getting issue only when there is any conversation from the MS Teams channel. 

![image](https://github.com/user-attachments/assets/7bff7d7f-d63d-4e87-bf7a-ac723268d6eb)

I am using the TeamsFx version of 2.3.1 and i tried to update it to 2.3.2 and 2.3.3 but still facing the issue. My Azure app node version was 20 and i tried with 18 also but no luck.

Please let me know if any further information is required. Any help in resolving issue is much appreciated.

**Comment by sayali-MSFT** [comment]
Hello @bhaskar-012024 , Thank you for your inquiry about your Teams app development issue! To assist you better, could you please provide the following details? 

Reproduction Steps: Please share the steps you took to encounter the issue. 

Documentation Links: Any specific documentation you followed that relates to your problem. 

Teams Client Version: What version of the Teams client are you using? 

Code Snippets: Relevant code snippets that illustrate the issue. 

This information will help us better understand the situation and provide a more accurate response.

**Comment by tecton** [comment]
Hi @bhaskar-012024, what kind of API have you used from @microsoft/teamsfx? E.g. notification?

**Comment by bhaskar-012024** [comment]
Hi @tecton ,
Yes, I am importing teamsfx and using it in below style as suggested in some of the examples inorder to send the proactive message.

const { BotBuilderCloudAdapter } = require('@microsoft/teamsfx');
const ConversationBot = BotBuilderCloudAdapter.ConversationBot;

const notificationApp = new ConversationBot({
    // See https://aka.ms/about-bot-adapter to learn more about adapters.
    adapter,
    // Enable notification
    notification: {
        enabled: true
    }
}
); 

const member1 = await notificationApp.notification.findMember(
    async (m) => {
        return m.parent.conversationReference.user.id === userId;
    }
);

**Comment by tecton** [comment]
Hi @bhaskar-012024, it might be caused by the corrupted cache file. Could you please delete/check the ".notification.localstore.json" file and try again?

**Comment by bhaskar-012024** [comment]
Yes @tecton , thank you for providing the resolution. Now it is working as expected. 
Is there anything I need to do in order to avoid this kind of issue or it is completely internal within the framework.

**Comment by tecton** [comment]
Hi @bhaskar-012024, this file is used as a default implementation to persist bot notification target references in the file. It's recommended to use your own store for production environment by extending interface `ConversationReferenceStore`. ([link](https://learn.microsoft.com/en-us/javascript/api/@microsoft/teamsfx/notificationoptions?view=msteams-client-js-latest#@microsoft-teamsfx-notificationoptions-storage)) We'll improve the error message for easier troubleshooting.

**Comment by bhaskar-012024** [comment]
Hi @tecton ,

I have gone through this documentation for  [ConversationReferenceStore](https://learn.microsoft.com/en-us/javascript/api/@microsoft/teamsfx/conversationreferencestore?view=msteams-client-js-latest) and its methods like add, list and remove. I understand like this is used to store the conversation reference of the user interacted with the bot and use it later to only send proactive message. 

Code structure where i am using notificationApp

const { BotBuilderCloudAdapter } = require('@microsoft/teamsfx');
const ConversationBot = BotBuilderCloudAdapter.ConversationBot;

const notificationApp = new ConversationBot({
// See https://aka.ms/about-bot-adapter to learn more about adapters.
adapter,
// Enable notification
notification: {
enabled: true
}
}
);

Case1 : To retrieve email
const member1 = await notificationApp.notification.findMember(
async (m) => {
return m.parent.conversationReference.user.id === userId;
}
);
const userEmail = member1.account.email;

Case 2: To identify user by email and send notification
const member = await notificationApp.notification.findMember(
    async (m) => {
        return m.account.email.toLowerCase() === <Email Address>
    }, SearchScope.Person
);

Then using member.parent.conversationReference proprty to get the conversation reference , user id, service url and sending proactive message using 
await connector.conversations.sendToConversation(conversationReferenceId, proactiveMessage);

Do i need to still use ConversationReferenceStore and store the details even if I was not using then to send the proactive message?
Can you please also share any teams sample example where we use ConversationReferenceStore, that helps me to implement it?
Thanks in advance.

**Comment by tecton** [comment]
Hi @bhaskar-012024, yes you need to use conversation reference store to find member. You can choose what properties to store when implement the interface.
Please refer to the following implementation in samples that leverage Azure storage.
1. https://github.com/OfficeDev/teams-toolkit-samples/blob/dev/large-scale-notification/src/internal/tableStore.ts
2. https://github.com/OfficeDev/teams-toolkit-samples/blob/dev/adaptive-card-notification/src/store/blobStore.ts
